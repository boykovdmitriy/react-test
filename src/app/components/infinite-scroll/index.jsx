import React from 'react';
import PropTypes from 'prop-types';

export class InfiniteScroll extends React.Component {
    constructor(props) {
        super(props);
        this.scrollEventListener = () => this.handleOnScroll();
    }

    componentDidMount() {
        window.addEventListener('scroll', this.scrollEventListener);
    }

    componentWillUnmount() {
        window.removeEventListener('scroll', this.scrollEventListener);
    }

    handleOnScroll() {
        const {handleScroll} = this.props;
        const scrollTop = (document.documentElement && document.documentElement.scrollTop) || document.body.scrollTop;
        const scrollHeight = (document.documentElement && document.documentElement.scrollHeight) || document.body.scrollHeight;
        const clientHeight = document.documentElement.clientHeight || window.innerHeight;
        const scrolledToBottom = Math.ceil(scrollTop + clientHeight) >= scrollHeight;

        if (scrolledToBottom) {
            handleScroll();
        }
    };

    render() {
        const {children} = this.props;
        return (
            <section>
                {children}
            </section>
        )
    }
}

InfiniteScroll.propTypes = {
    children: PropTypes.object.isRequired,
    handleScroll: PropTypes.func.isRequired
};