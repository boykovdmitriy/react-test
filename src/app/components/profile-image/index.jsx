import React from 'react';
import PropTypes from 'prop-types'
import style from './style.css'

export class ProfileImage extends React.Component {
    render() {
        const {thumb} = this.props;
        return (
            <img className={style.image_preview} src={thumb}/>
        )
    }
}

ProfileImage.propTypes = {
    thumb: PropTypes.string.isRequired,
    origin: PropTypes.string.isRequired
};