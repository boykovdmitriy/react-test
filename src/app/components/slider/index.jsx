import React from 'react';
import {Carousel} from 'react-responsive-carousel';

export const WrappedSlider = ({children}) => {
    const settings = {
        showArrows: true,
        showThumbs: false,
        infiniteLoop: true,
        autoPlay: true,
        interval: 1000,
        centerMode: true,
        showIndicators: false,
        showStatus: false
    };

    return (
        <Carousel {...settings}>
            {children}
        </Carousel>
    );
};