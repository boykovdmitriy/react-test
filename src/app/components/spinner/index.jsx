import React from 'react'
import style from './spinner.css';

export const Spinner = () => (
    <div className={style.loader}/>
);