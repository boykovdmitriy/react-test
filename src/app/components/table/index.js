export * from './Table';
export * from './table-column';
export * from './table-row';
