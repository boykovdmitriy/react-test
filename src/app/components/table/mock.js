export default [{
    id: 1,
    name: 'first',
    website: "http://mohr.co",
    profile_image_urls: {
        original: "http://s3-eu-west-1.amazonaws.com/staging-salony/image/profiles/files/000/000/064/original/1.jpg?1448538378",
        thumb: "http://s3-eu-west-1.amazonaws.com/staging-salony/image/profiles/files/000/000/064/thumb/1.jpg?1448538378"
    }
}, {
    id: 1,
    name: 'second',
    website: "http://mohr.co",
    profile_image_urls: {
        original: "http://s3-eu-west-1.amazonaws.com/staging-salony/image/profiles/files/000/000/064/original/1.jpg?1448538378",
        thumb: "http://s3-eu-west-1.amazonaws.com/staging-salony/image/profiles/files/000/000/064/thumb/1.jpg?1448538378"
    }
}];