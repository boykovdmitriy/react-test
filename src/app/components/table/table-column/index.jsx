import React from 'react';
import style from '../table.css'

export const TableColumn = ({children}) => (
    <td className={style.column}>
        {children}
    </td>
);