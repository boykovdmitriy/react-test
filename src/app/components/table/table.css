.table {
    border-collapse: collapse;
    width: 100%;
}

.header, .column {
    padding: 8px;
    text-align: left;
    border-bottom: 1px solid #ddd;
    max-height: 5rem;
}

.row:hover {background-color:#f5f5f5;}