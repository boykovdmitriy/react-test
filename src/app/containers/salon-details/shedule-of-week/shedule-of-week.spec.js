import React from 'react';
import {SheduleOfWeek} from './index'
import {TableColumn} from "components/table";

describe('SheduleOfWeek', () => {
    it('render SheduleOfWeek component', () => {
        const hours = {
            fri:
                {to: "20:00", from: "07:00"},
            mon:
                {to: "20:00", from: "07:00"},
            sat:
                {to: "20:00", from: "07:00"},
            sun:
                {to: "20:00", from: "07:00"},
            thu:
                {to: "21:00", from: "07:00"},
            wed:
                {to: "20:00", from: "07:00"}
        };
        const wrapper = shallow(
            <SheduleOfWeek workHours={hours}/>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<TableColumn key='tue'>-</TableColumn>).length).toEqual();
    });
    it('render SheduleOfWeek component without a required field', () => {
        expect(() => shallow(<SheduleOfWeek/>)).toThrow();
    });
});