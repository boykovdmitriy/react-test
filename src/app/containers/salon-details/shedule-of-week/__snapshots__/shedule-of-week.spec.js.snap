// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SheduleOfWeek render SheduleOfWeek component 1`] = `
<Table
  headers={
    Array [
      "sun",
      "mon",
      "tue",
      "wed",
      "thu",
      "fri",
      "sat",
    ]
  }
>
  <TableRow>
    <TableColumn
      key="sun"
    >
      07:00-20:00
    </TableColumn>
    <TableColumn
      key="mon"
    >
      07:00-20:00
    </TableColumn>
    <TableColumn
      key="tue"
    >
      -
    </TableColumn>
    <TableColumn
      key="wed"
    >
      07:00-20:00
    </TableColumn>
    <TableColumn
      key="thu"
    >
      07:00-21:00
    </TableColumn>
    <TableColumn
      key="fri"
    >
      07:00-20:00
    </TableColumn>
    <TableColumn
      key="sat"
    >
      07:00-20:00
    </TableColumn>
  </TableRow>
</Table>
`;
